recipes/eleuther_eval.py

```diff
@@ -4,40 +4,279 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+import importlib.metadata
 import sys
 import time
 
-from typing import Any, Dict, List, Tuple, Union
+from typing import Dict, List, Tuple, Union
+
+import PIL
 
 import torch
 from omegaconf import DictConfig
 
-from torch import nn
-from torchtune import config, generation, training, utils
-from torchtune.data import left_pad_sequence
+from torchtune import config, training, utils
+from torchtune.data import (
+    format_content_with_images,
+    left_pad_sequence,
+    Message,
+    padded_collate_tiled_images_and_mask,
+)
+from torchtune.generation import generate, sample
 from torchtune.modules import TransformerDecoder
+from torchtune.modules.model_fusion import DeepFusionModel
 from torchtune.modules.tokenizers import ModelTokenizer
+from torchtune.modules.transforms import Transform
 from torchtune.recipe_interfaces import EvalRecipeInterface
 
-logger = utils.get_logger("DEBUG")
-
 try:
-    import lm_eval
-    from lm_eval.evaluator import evaluate, get_task_list
-    from lm_eval.models.huggingface import HFLM
-    from lm_eval.tasks import get_task_dict, TaskManager
-    from lm_eval.utils import make_table
-except ImportError:
-    logger.error(
-        "Recipe requires EleutherAI Eval Harness v0.4. Please install with `pip install lm_eval==0.4.*`"
+    lm_eval_version = importlib.metadata.version("lm_eval")
+except importlib.metadata.PackageNotFoundError:
+    lm_eval_version = ""
+
+if not lm_eval_version >= "0.4.2":
+    raise ImportError(
+        "lm_eval version must be >= 0.4.2. Please install lm_eval >= 0.4.2."
```

Comment:
fyi, i had to create this import_guard.py for another PR, because i was using it in multiple files. I wonder if it would make sense to move this there. My first intuition is that we should NOT, as keeping this logic closer to the code is better, unless you are using it somewhere else too: https://github.com/pytorch/torchtune/blob/main/torchtune/utils/_import_guard.py

Reply:
This is good to keep in mind - my guess is that we will end up utilizing version and package guards much more frequently as our recipes expand.